crypto/rsa/rsa_backend.c

```diff
@@ -137,7 +139,7 @@ int ossl_rsa_todata(RSA *rsa, OSSL_PARAM_BLD *bld, OSSL_PARAM params[])
         goto err;
 
     /* Check private key data integrity */
-    if (rsa_d != NULL) {
+    if (include_private && rsa_d != NULL) {
```

Comment:
I am not sure if this logic is right or not ?? - should having no private selection remove the factors/exp and coeff's?

Reply:
IMO this is correct for sure. These are (albeit optional) parts of a RSA private key.